scalene/scalene_sigqueue.py

```diff
@@ -14,15 +15,19 @@ def get(self):
         return self.queue.get()
 
     def start(self):
+        assert self.thread == None
         self.thread = threading.Thread(target=self.run, daemon=True)
         self.thread.start()
 
     def stop(self):
-        self.queue.put(None)
+        if self.thread != None:
+            self.queue.put(None)
+            self.thread.join()
+            self.thread = None
 
     def run(self):
         while True:
             item = self.queue.get()
-            if not item:
+            if item == None:  # None == stop request
```

Comment:
It's "Pythonic" to treat falsy things like `None` as bools.

Reply:
I thought checking for `None` explicitly was clearer, as we're looking for a specific value, rather than simply "anything that evaluates `False`"

Reply:
Any actual value will be truthy, so I don't think this is a problem. In principle, we should use a type annotation for this. But it's fine.